src/mysql.py

```diff
@@ -19,10 +23,32 @@ class MySQL:
     def __init__(self, conn):
         self.conn = conn
 
+    def get_global(self, cursor, var):
+        # "This statement does not require any privilege", so no need to handle permissions issues.
+        # https://dev.mysql.com/doc/refman/5.7/en/show-variables.html
+        cursor.execute("SELECT @@GLOBAL.%s", (var,))
+        return cursor.fetchone()[0]
+
+    def set_global(self, cursor, var, val, alt_value=None):
+        current = self.get_global(cursor, var)
+        if current == val or (alt_value and current == alt_value):
+            print("{} already set to {}.".format(var, val))
+        else:
+            print("Setting {}={}...".format(var, val))
+            try:
+                # http://stackoverflow.com/a/10077141/358804
+                cursor.execute("SET GLOBAL {} = {}".format(var, val))
+            except pymysql.err.DatabaseError as err:
+                code = err.args[0]
+                if code == pymysql.constants.ER.SPECIFIC_ACCESS_DENIED_ERROR:
+                    print("Unable to set {}={} in MySQL - you will need to do so externally.".format(var, val), file=sys.stderr)
+                else:
+                    raise
+
     def enable_logs(self):
         with self.conn.cursor() as cursor:
```

Comment:
Good eye! This is a funky/confusing edge case where GitHub shows a diff against the parent commit in the target branch (`master`), rather than the _latest_ of the target branch. In other words, the diff was showing out-of-date information. Rebased this branch, so that it now shows the proper diff.